test/Grpc.Net.Client.Tests/GrpcChannelTests.cs

```diff
@@ -29,66 +32,187 @@ public class GrpcChannelTests
         {
-            // Arrange & Assert
+            // Arrange & Act
             var ex = Assert.Throws<InvalidOperationException>(() => GrpcChannel.ForAddress("https://localhost", new GrpcChannelOptions
             {
                 Credentials = ChannelCredentials.Insecure
             }));
 
-            // Act
+            // Assert
             Assert.AreEqual("Channel is configured with insecure channel credentials and can't use a HttpClient with a 'https' scheme.", ex.Message);
         }
+
+        [Test]
+        public void Dispose_NotCalled_NotDisposed()
+        {
+            // Arrange
+            var channel = GrpcChannel.ForAddress("https://localhost");
+
+            // Act (nothing)
+
+            // Assert
+            Assert.IsFalse(channel.Disposed);
+        }
+
+        [Test]
+        public void Dispose_Called_Disposed()
+        {
+            // Arrange
+            var channel = GrpcChannel.ForAddress("https://localhost");
+
+            // Act
+            channel.Dispose();
+
+            // Assert
+            Assert.IsTrue(channel.Disposed);
+            Assert.Throws<ObjectDisposedException>(() => channel.HttpClient.CancelPendingRequests());
+        }
+
+        [Test]
+        public void Dispose_CalledMultipleTimes_Disposed()
+        {
+            // Arrange
+            var channel = GrpcChannel.ForAddress("https://localhost");
+
+            // Act
+            channel.Dispose();
+            channel.Dispose();
+
+            // Assert
+            Assert.IsTrue(channel.Disposed);
+        }
+
+        [Test]
+        public void Dispose_CreateCallInvoker_ThrowError()
+        {
+            // Arrange
+            var channel = GrpcChannel.ForAddress("https://localhost");
+
+            // Act
+            channel.Dispose();
+
+            // Assert
+            Assert.Throws<ObjectDisposedException>(() => channel.CreateCallInvoker());
+        }
+
+        [Test]
+        public async Task Dispose_StartCallOnClient_ThrowError()
+        {
+            // Arrange
+            var channel = GrpcChannel.ForAddress("https://localhost");
+            var client = new Greet.Greeter.GreeterClient(channel);
+
+            // Act
+            channel.Dispose();
+
+            // Assert
+            await ExceptionAssert.ThrowsAsync<ObjectDisposedException>(() => client.SayHelloAsync(new Greet.HelloRequest()).ResponseAsync);
+        }
+
+        public class TestHttpMessageHandler : HttpMessageHandler
+        {
+            public bool Disposed { get; private set; }
+
+            protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+            {
+                throw new NotImplementedException();
+            }
+
+            protected override void Dispose(bool disposing)
+            {
+                Disposed = true;
+            }
+        }
+
+        [Test]
+        public void Dispose_CalledWhenHttpClientSpecified_Disposed()
```

Comment:
nit: shouldn't the name be `NotCalled`. This test is verifying that an HttpClient is specified, but DisposeHttpClient is not set then dispose is not called?